torch_geometric/explain/algorithm/pgm_explainer.py

```diff
@@ -0,0 +1,462 @@
+            p_values.append(p)
+            if p < significance_threshold:
+                dependent_nodes.append(node)
+
+        top_p = np.min((max_subgraph_size, num_nodes - 1))
+        ind_top_p = np.argpartition(p_values, top_p)[0:top_p]
+        pgm_nodes = list(ind_top_p)
+
+        node_mask = torch.zeros(x.size(), dtype=torch.int)
+        node_mask[pgm_nodes] = 1
+        pgm_stats = torch.tensor(p_values)
+
+        return node_mask, pgm_stats
+
+    def explain_node(
+        self,
+        model: torch.nn.Module,
+        x: torch.Tensor,
+        index: int,
+        edge_index: torch.Tensor,
+        edge_weight: torch.Tensor,
+        target: torch.Tensor,
+        num_samples: int = 100,
+        max_subgraph_size: int = None,
+        significance_threshold: float = 0.05,
+        pred_threshold: float = 0.1,
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        r"""Generate explanations for node classification tasks
+
+        Args:
+            model (torch.nn.Module): model that generated the predictions
+            x (torch.Tensor): node feature matrix
+            index (torch.Tensor): the index of the node that the
+                explanations are being generated for
+            edge_index (torch.Tensor): edge_index of the input graph
+            edge_weight (torch.Tensor): apply weighting to edges (Optional)
+            target (torch.Tensor):  the prediction labels
+            num_samples (int): num of samples of perturbations used to test
+                the significance of neighbour nodes to the prediction
+            max_subgraph_size:
+            significance_threshold (float): threshold below which to consider
+                a node has an effect on the prediction
+            pred_threshold:
+
+        Returns:
+            node_mask (torch.Tensor): 1 or 0 corresponding to whether a node
+                is significant in the selected node's prediction
+            pgm_stats (torch.Tensor): p-values of all the nodes in the
+                graph ordered by node index
+        """
+
+        import pandas as pd
+        from pgmpy.estimators.CITests import chi_square
+
+        logging.info(f'Explaining node: {index}')
+
+        neighbors, edge_index_new, mapping, edge_mask_new = k_hop_subgraph(
+            node_idx=index,
+            num_hops=get_num_hops(model),
+            edge_index=edge_index,
+            relabel_nodes=False,
+            num_nodes=x.size(0),
+        )
+
+        if index not in neighbors:
+            neighbors = torch.cat([neighbors, index], dim=1)
+
+        pred_model = model(x, edge_index, edge_weight)
+
+        softmax_pred = torch.softmax(pred_model, dim=1)
+
+        samples = []
+        pred_samples = []
+
+        for iteration in range(num_samples):
+            # a subset of neighbours are selected randomly for perturbing
+            seeds = np.random.choice([1, 0], size=(len(neighbors), ))
+            X_perturb = self.perturb_features_on_nodes(
```

Comment:
```suggestion
            x_perturb = self.perturb_features_on_nodes(
```
lets keep variable names lowercase.